components/service/glean/src/main/java/mozilla/components/service/glean/scheduler/BaselinePingScheduler.kt

```diff
@@ -0,0 +1,135 @@
+import mozilla.components.support.base.log.logger.Logger
+import java.util.concurrent.TimeUnit
+
+/**
+ * BaselinePingScheduler facilitates sending and scheduling of baseline pings when the application
+ * enters the background state (see [GleanLifecycleObserver]).  Using a [PING_INTERVAL_HOURS] of
+ * zero means that there is no restriction on how many pings that can be sent per time period.
+ */
+internal class BaselinePingScheduler(val applicationContext: Context) {
+    private val logger = Logger("glean/BaselinePingScheduler")
+    internal val sharedPreferences: SharedPreferences by lazy { getSharedPrefs() }
+
+    companion object {
+        const val LAST_PING_TIMESTAMP_KEY = "last_baseline_ping_timestamp"
+        const val STORE_NAME = "baseline"
+        const val PING_INTERVAL_HOURS = 0L
+    }
+
+    init {
+        // Set the OS type
+        GleanBaseline.os.set("Android")
+
+        // Set the OS version
+        // https://developer.android.com/reference/android/os/Build.VERSION
+        GleanBaseline.osVersion.set(Build.VERSION.SDK_INT.toString())
+
+        // Set the device strings
+        // https://developer.android.com/reference/android/os/Build
+        GleanBaseline.deviceManufacturer.set(Build.MANUFACTURER)
+        GleanBaseline.deviceModel.set(Build.MODEL)
+
+        // Set the CPU architecture
+        GleanBaseline.architecture.set(Build.SUPPORTED_ABIS[0])
+
+        // Set the enabled accessibility services
+        getEnabledAccessibilityServices(
+            applicationContext.getSystemService(Context.ACCESSIBILITY_SERVICE) as AccessibilityManager
+        ) ?.let {
+            GleanBaseline.a11yServices.set(it)
+        }
+    }
+
+    /**
+     * Check to see if a ping can be sent.
+     *
+     * This will compare the current system time to the last ping time stored in [SharedPreferences]
+     * to determine whether [PING_INTERVAL_HOURS] has elapsed.
+     *
+     * @return true if the time interval has elapsed or if this is the first attempt to send a
+     *         ping, otherwise false.
+     */
+    fun canSendPing(): Boolean {
+        val lastPingTime = try {
+            sharedPreferences.getLong(LAST_PING_TIMESTAMP_KEY, 0L)
+        } catch (e: ClassCastException) {
+            // Intentionally doing nothing here, if LAST_PING_TIMESTAMP_KEY's value is
+            // not a Long, then we will let lastPingTime be 0L for the purposes of comparison in
+            // order to allow the ping to be sent immediately
+            logger.error("BaselinePingScheduler last ping time stored was not a Long value")
+            0L
+        }
+
+        val currentTime = System.currentTimeMillis()
+
+        if (lastPingTime == 0L ||
+            currentTime - lastPingTime >= TimeUnit.HOURS.toMillis(PING_INTERVAL_HOURS)) {
+            return true
+        }
+
+        return false
+    }
+
+    /**
+     * Update the persisted timestamp when the metrics ping is sent.
```

Comment:
`metrics` -> `baseline` ?  (Or am I misunderstanding?)

Reply:
Nope, just missed updating that particular comment.